src/Microsoft.AspNet.Hosting/Internal/RequestServicesContainerMiddleware.cs

```diff
@@ -13,18 +15,40 @@ public class RequestServicesContainerMiddleware
         private readonly RequestDelegate _next;
         private readonly IServiceProvider _services;
 
-        public RequestServicesContainerMiddleware(RequestDelegate next, IServiceProvider services)
+        public RequestServicesContainerMiddleware([NotNull] RequestDelegate next, [NotNull] IServiceProvider services)
         {
             _services = services;
             _next = next;
         }
 
-        public async Task Invoke(HttpContext httpContext)
+        public async Task Invoke([NotNull] HttpContext httpContext)
         {
-            using (var container = RequestServicesContainer.EnsureRequestServices(httpContext, _services))
+            // All done if there request services is set
+            if (httpContext.RequestServices != null)
             {
-                await _next.Invoke(httpContext);
+                return;
```

Comment:
Oops yeah, I should add a test that exercises the request services already being populated and not used